web_monitoring/diffing_server.py

```diff
@@ -252,19 +250,20 @@ def fetch_diffable_content(self, url, expected_hash, query_params):
                                 reason=(f'Fetched content at "{url}" does not '
                                         f'match hash "{expected_hash}".'))
 
-        raise tornado.gen.Return(response)
+        return response
 
-    @tornado.gen.coroutine
-    def diff(self, func, a, b, params, tries=2):
+    async def diff(self, func, a, b, params, tries=2):
         """
         Actually do a diff between two pieces of content, optionally retrying
         if the process pool that executes the diff breaks.
         """
         executor = self.get_diff_executor()
         for attempt in range(tries):
             try:
-                result = yield executor.submit(caller, func, a, b, **params)
-                raise tornado.gen.Return(result)
+                loop = asyncio.get_running_loop()
```

Comment:
You could probably pull this out of the for loop and do it at the same time we first get `executor`.